src/Components/Aspire.Confluent.Kafka/AspireKafkaConsumerExtensions.cs

```diff
@@ -0,0 +1,175 @@
+        AddKafkaConsumer(builder, $"{DefaultConfigSectionName}:{name}", configureSettings, configureBuilder, connectionName: name, serviceKey: name);
+    }
+
+    private static void AddKafkaConsumer<TKey, TValue>(
+        IHostApplicationBuilder builder,
+        string configurationSectionName,
+        Action<KafkaConsumerSettings>? configureSettings,
+        Action<ConsumerBuilder<TKey, TValue>>? configureBuilder,
+        string connectionName,
+        string? serviceKey)
+    {
+        ArgumentNullException.ThrowIfNull(builder);
+
+        var settings = BuildConsumerSettings(builder, configurationSectionName, configureSettings, connectionName);
+
+        if (serviceKey is null)
+        {
+            builder.Services.AddSingleton<ConsumerConnectionFactory<TKey, TValue>>(sp => CreateConsumerConnectionFactory<TKey, TValue>(sp, configureBuilder, settings));
+            builder.Services.AddSingleton<IConsumer<TKey, TValue>>(sp => sp.GetRequiredService<ConsumerConnectionFactory<TKey, TValue>>().Create());
+        }
+        else
+        {
+            builder.Services.AddKeyedSingleton<ConsumerConnectionFactory<TKey, TValue>>(serviceKey, (sp, key) => CreateConsumerConnectionFactory<TKey, TValue>(sp, configureBuilder, settings));
+            builder.Services.AddKeyedSingleton<IConsumer<TKey, TValue>>(serviceKey, (sp, key) => sp.GetRequiredKeyedService<ConsumerConnectionFactory<TKey, TValue>>(key).Create());
+        }
+
+        if (settings.Metrics)
+        {
+            builder.Services.TryAddSingleton<MetricsChannel>();
+            builder.Services.AddHostedService<MetricsService>();
+            builder.Services.TryAddSingleton<ConfluentKafkaMetrics>();
+            builder.Services.AddOpenTelemetry().WithMetrics(metricBuilderProvider => metricBuilderProvider.AddMeter(ConfluentKafkaCommon.MeterName));
+        }
+
+        if (settings.HealthChecks)
+        {
+            string healthCheckName = serviceKey is null
+                ? ConfluentKafkaCommon.ConsumerHealthCheckName
+                : string.Format(CultureInfo.InvariantCulture, ConfluentKafkaCommon.KeyedConsumerHealthCheckNameFormat, connectionName);
+
+            builder.TryAddHealthCheck(new HealthCheckRegistration(healthCheckName,
+                sp =>
+                {
+                    var connectionFactory = serviceKey is null
+                        ? sp.GetRequiredService<ConsumerConnectionFactory<TKey, TValue>>()
+                        : sp.GetRequiredKeyedService<ConsumerConnectionFactory<TKey, TValue>>(serviceKey);
+
+                    var options = new KafkaHealthCheckOptions();
+                    options.Configuration = new ProducerConfig(connectionFactory.Config.ToDictionary());
+                    options.Configuration.SocketTimeoutMs = 1000;
+                    options.Configuration.MessageTimeoutMs = 1000;
+                    options.Configuration.StatisticsIntervalMs = 0;
+                    return new KafkaHealthCheck(options);
```

Comment:
Instead of establishing a new connection, should the health check use the IProducer we registered in DI? See 

https://github.com/dotnet/aspire/blob/23e168d898f8ab8ad3eaafa2db7d6af47652dd78/src/Components/README.md?plain=1#L96

Reply:
The `AspNetCore.HealthChecks.Kafka` package builds its own producer instance. 
If it was implemented, another limitation would be at consumer side we would not be able to reuse the client instance as these are different instances.